internal/orchestration/controller.go

```diff
@@ -329,34 +208,43 @@ func (e *Experiment) writeAssignmentsFile() (err error) {
 	}()
 	enc := json.NewEncoder(f)
 	enc.SetIndent("", "\t")
+
+	replicaIDs := make(map[string][]hotstuff.ID)
+	for host, opts := range m {
+		replicaIDs[host] = make([]hotstuff.ID, 0, len(opts))
+		for _, opt := range opts {
+			replicaIDs[host] = append(replicaIDs[host], opt.HotstuffID())
+		}
+	}
+
 	return enc.Encode(assignmentsFileContents{
-		HostsToReplicas: e.hostsToReplicas,
-		HostsToClients:  e.hostsToClients,
+		HostsToReplicas: replicaIDs,
+		HostsToClients:  clientIDs,
```

Comment:
Reminder: Consider what needs to be written to the `hosts.json` file, and should this be a more comprehensive format with more experiment details.